job/job.go

```diff
@@ -34,6 +35,34 @@ type Job struct {
 	instance *packet.Instance
 }
 
+var tinkWorkflowClient = client.GetWorkflowsFromTink
+
+// discoverWorkflowForMac fetches the workflow for a given MAC
+// hwID is the hardware/worker ID corresponding to the MAC
+func discoverActiveWorkflowForMac(hwID string) (bool, error) {
+	var active bool = false
+
+	s := fmt.Sprintf("fetching workflows for hardware %s", hwID)
+	joblog.With("dhcp", hwID).Info(s)
+
+	wcl, err := tinkWorkflowClient(hwID)
+	if err != nil {
+		return false, err
+	}
+
+	// wcl is workflowContextList, for more details refer to workflow.pb.go in tink
+	for _, wf := range wcl.WorkflowContexts {
+		if wf.CurrentActionState == 0 {
+			active = true
+			s := fmt.Sprintf("found active workflow for hardware %s with workflow "+
+				"id %s", hwID, wf.WorkflowId)
+			joblog.With("dhcp", hwID).Info(s)
```

Comment:
these args should really be kv pairs in the log line like:

```go
l.With("worflowID", wf.WorkflowId).Info("found active workflow for hardware")
```

Reply:
done

Reply:
done